.github/workflows/main-workflow.yml

```diff
@@ -18,8 +18,7 @@ jobs:
     - name: setup and run linter
       run: |
         pip install cpplint
-        cpplint  --filter=-legal,-build/c++11 --linelength=100 *.cpp *.h
-
+        cpplint  --filter=-legal,-build/c++11,-build/include_what_you_use,-whitespace/braces,-whitespace/operators,-whitespace/parens,-whitespace/indent,-readability/braces --linelength=100 *.cpp *.h

```

Comment:
Good to customize the linter as per your need :)
But why disable `build/include_what_you_use`? Would you need includes that are not used as well?